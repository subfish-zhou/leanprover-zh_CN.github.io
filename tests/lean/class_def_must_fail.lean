/- The following definition should fail. -/
@[class] def Foo (n : Nat) : Prop := n > 2


def Bla (n : Nat) : Prop := n > 2

attribute [class] Bla
