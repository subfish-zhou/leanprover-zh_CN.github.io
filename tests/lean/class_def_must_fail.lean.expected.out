class_def_must_fail.lean:2:13-2:16: error: invalid 'class', declaration 'Foo' must be inductive datatype or structure
class_def_must_fail.lean:7:18-7:21: error: invalid 'class', declaration 'Bla' must be inductive datatype or structure
